janitor/functions/conditional_join.py

```diff
@@ -578,7 +605,12 @@ def _less_than_indices(
         if not search_indices.size:
             return None
     if right_is_sorted and (keep == _KeepTypes.FIRST.value):
-        return left_index, right_index[search_indices]
+        if any_nulls:
+            return left_index, right_index[search_indices]
```

Comment:
@samukweku this in particular feels like an important thing to add to the test suite; would you concur?

Reply:
Sure ... I'll add a test for it